SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex

```diff
@@ -1654,6 +1657,54 @@ \section{Initialization and Assignment}
 example \code{FixedArray::m\_InternalArray} and \code{Index::m\_InternalArray}
 do not have a default member initializer.
 
+\subsection{Initializing variables of fixed size array types}
+\label{subsec:InitializingVariablesOfFixedSizeArrayTypes}
+
+ITK has various fixed size array types, including template instantiations of
+\code{Index}, \code{Size}, \code{FixedArray}, \code{Point}, and \code{Vector}.
+
+A variable of such a fixed size array type can be zero-initialized by an empty
+initializer list, `{}`. This is usually the preferred way to initialize the
+variable, when it should initially be filled with zeroes. For example:
+
+\small
+\begin{minted}[baselinestretch=1,fontsize=\footnotesize,linenos=false,bgcolor=ltgray]{cpp}
+// Declare an index at position (0, 0).
+Index<2> index{};
+
+// Declare a point whose coordinates are all 0.0 (the origin).
+PointType origin{};
+\end{minted}
+\normalsize
+
+\code{Index} and \code{Size} both have a static `Filled(fillValue)` member
+function, to allow creating a variable that is filled with an arbitrary value.
+For these types, this is usually the preferred way to initialize the variable,
+when it should initially be filled with a value that may be non-zero. For
+example:
+
+\small
+\begin{minted}[baselinestretch=1,fontsize=\footnotesize,linenos=false,bgcolor=ltgray]{cpp}
+// Declare the index {1, 1, 1}.
+auto index = Index<3>::Filled(1);
+
+// Declare the size of an 256 x 256 image.
+auto imageSize = Size<2>::Filled(256);
+\end{minted}
+\normalsize
+
+For other fixed size array types, the function `itk::MakeFilled<T>(fillValue)`
+is preferably, when the array should initially be filled with a value that may
+be non-zero. For example:
+
+\small
+\begin{minted}[baselinestretch=1,fontsize=\footnotesize,linenos=false,bgcolor=ltgray]{cpp}
+// Declare a spacing filled with the value 1.0 (for each direction).
```

Comment:
Spacing of 1.0 is the default, so maybe choose another value here, such as 0.5 or 1.2.

Reply:
Well... A spacing of 1.0 is the default for `itk::Image`, as well as for various image filters. But 1.0 is usually _not_ the default for `SpacingType`, because  `SpacingType` is typically defined as `Vector<double, Dimension>`. (Which is zero-filled, "by default".) So `spacing = MakeFilled<SpacingType>(1.0)` _does_ make sense to me. And the ITK source tree does still have quite a few of such `MakeFilled<SpacingType>(1.0)` calls, even after the clean-up by Hans (@hjmjohnson).

Nevertheless, if you insist, I'll make it `0.5` instead, for you. It's no big deal. 😃 Would you like that?

Reply:
I am aware that 0 is default for SpacingType, and that 1.0 is commonly used (and default for Image). I would still like for this to be something other than those two numbers, e.g. `0.5`.

Reply:
OK, please check: [force-push](https://github.com/InsightSoftwareConsortium/ITKSoftwareGuide/compare/5e5f2e577ff73229e6162c683210c407cf369cce..f10846cff4a02eb5d60a2e62c2238965fefd9533)

By the way, for `1.2` I think would add a constant, to avoid [readability-magic-numbers](https://clang.llvm.org/extra/clang-tidy/checks/readability/magic-numbers.html) warnings, theoretically. 

    constexpr auto spacingValue = 1.2;
    spacing = MakeFilled<SpacingType>(spacingValue);

But I think `0.5` is not considered a magic number, fortunately 😇  .